### PR TITLE
Fixed composer & empty line

### PR DESCRIPTION
**Pull request by Madlexx**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,69 +1,63 @@
...
-    "autoload": {
-        "psr-4": {
-            "": "src/"
-        },
-        "classmap": [
-            "app/AppKernel.php",
-            "app/AppCache.php"
-        ]
+  "name": "alex/test",
+  "license": "proprietary",
```
Comment (aivus) [inline composer.json]: use `MIT` license

File: composer.json [hunk 2]
```diff
@@ -1,69 +1,63 @@
 {
-    "name": "alex/test",
-    "license": "proprietary",
-    "type": "project",
-    "autoload": {
```
Comment (aivus) [inline composer.json]: Why are you reformat file?

Comment (Madlexx) [inline composer.json]: It's bug with my Phpstorm settings(indent for json equals 2 not 4).

Comment (aivus) [inline composer.json]: Please, change indent settings to 4

File: src/Application/BlogBundle/Form/Type/ThumbnailType.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class ThumbnailType extends AbstractType
+{
+
```
Comment (aivus) [inline src/Application/BlogBundle/Form/Type/ThumbnailType.php]: Учекф дшту

File: src/Application/BlogBundle/Admin/PostAdmin.php [hunk 4]
```diff
@@ -1,5 +1,6 @@
 <?php namespace Application\BlogBundle\Admin;
 
+use Application\BlogBundle\Form\Type\ThumbnailType;
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/PostAdmin.php]: Group this

File: src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<?php namespace Application\BlogBundle\Form\DataTransformer;
+
+use Application\BlogBundle\Service\Thumbnail;
+use Symfony\Component\Form\DataTransformerInterface;
```
Comment (aivus) [inline src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php]: Group this

File: src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+class UploadedFileToThumbnail implements DataTransformerInterface
+{
+    /** @var Thumbnail */
+    private $thumbnail;
+
+    public function __construct(Thumbnail $thumbnail)
+    {
+        $this->thumbnail = $thumbnail;
+    }
+    /** {@inheritdoc} */
```
Comment (aivus) [inline src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php]: Please, add new line before this line

File: src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php (hunk 5 above)
Comment (aivus) [inline src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php]: Group this

File: src/Application/BlogBundle/Form/Type/ThumbnailType.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        $view->vars['accept']  = isset($options['accept']) ? $options['accept'] : 'image/*';
+
+
+        $view->vars['image_url'] = $view->vars['data'] ?: null;
+        $view->vars = array_replace($view->vars, array(
+            'type' => 'file',
+            'value' => '',
+        ));
+    }
+    /**
```
Comment (aivus) [inline src/Application/BlogBundle/Form/Type/ThumbnailType.php]: Please, add new line before this line

File: src/Application/BlogBundle/Resources/config/config.yml [hunk 8]
```diff
@@ -8,6 +8,8 @@ twig:
     form:
         resources:
             - 'SonataFormatterBundle:Form:formatter.html.twig'
+            - 'ApplicationBlogBundle:Form:fields.html.twig'
+
```
Comment (aivus) [inline src/Application/BlogBundle/Resources/config/config.yml]: Too many empty lines

File: src/Application/BlogBundle/Resources/config/services.yml [hunk 9]
```diff
@@ -9,3 +9,17 @@ services:
         arguments: [~, Application\BlogBundle\Entity\Category, ~]
         tags:
             - { name: sonata.admin, manager_type: orm, group: 'blog', label: 'blog_bundle.admin.category.singular' }
+    application.blog_thumbnail:
```
Comment (aivus) [inline src/Application/BlogBundle/Resources/config/services.yml]: please, add new empty line

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        return (string) self::UPLOAD_DIRECTORY .'/'. $file->getClientOriginalName();
+    }
+
+    /**
+     * Set folder for file uploads.
+     */
+    protected function setFolder()
+    {
+        $this->folder = getcwd().self::UPLOAD_DIRECTORY;
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Upload folder should be setted in config

Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Upload folder should be set in config

Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Upload folder should be set in config

File: src/Application/BlogBundle/Form/Type/ThumbnailType.php [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'thumbnail';
+    }
+
```
Comment (aivus) [inline src/Application/BlogBundle/Form/Type/ThumbnailType.php]: excessive line

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @return bool|\Exception|FileException
+     */
+    public function save(UploadedFile $uploadedFile)
+    {
+        $file = $uploadedFile;
+        try {
+            $file->move($this->folder, $file->getClientOriginalName());
+        } catch (FileException $e) {
+            return $e;
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: please, return `false` if exception happens

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * @param UploadedFile $uploadedFile
+     *
+     * @return bool|\Exception|FileException
+     */
+    public function save(UploadedFile $uploadedFile)
+    {
+        $file = $uploadedFile;
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Why are you creating variable `$file`? You can use `$uploadedFile`

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class Thumbnail
+{
+    protected $folder;
+    const UPLOAD_DIRECTORY = '/upload/thumbnails';
+
+    public function __construct()
+    {
+        $this->setFolder();
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: You don't need to set a folder.
You should pass this folder via DI to constructor and use it.

File: src/Application/BlogBundle/Service/Thumbnail.php (hunk 10 above)
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Also, don't forget to put spaces around `.`

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Create folder by path.
+     *
+     * @return bool|\Exception
+     */
+    protected function createFolderIfNotExists()
+    {
+        $this->createFolderIfNotExists();
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: ???
You call this function inside itself.

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        $this->folder = getcwd().self::UPLOAD_DIRECTORY;
+    }
+
+    /**
+     * Create folder by path.
+     *
+     * @return bool|\Exception
+     */
+    protected function createFolderIfNotExists()
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: This function isn't used in class

Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: This function isn't used in class

File: src/Application/BlogBundle/Resources/config/doctrine/Post.orm.yml [hunk 17]
```diff
@@ -25,6 +25,7 @@ Application\BlogBundle\Entity\Post:
         description:
             type: string
             length: '255'
```
Comment (aivus) [inline src/Application/BlogBundle/Resources/config/doctrine/Post.orm.yml]: Please, don't use `'` for integers

File: src/Application/BlogBundle/Resources/config/doctrine/Post.orm.yml [hunk 18]
```diff
@@ -25,6 +25,7 @@ Application\BlogBundle\Entity\Post:
...
+            nullable: true
         keywords:
             type: string
             length: '255'
```
Comment (aivus) [inline src/Application/BlogBundle/Resources/config/doctrine/Post.orm.yml]: Please, don't use `'` for integers

File: src/Application/BlogBundle/Admin/CategoryAdmin.php [hunk 19]
```diff
@@ -1,4 +1,5 @@
-<?php namespace Application\BlogBundle\Admin;
+<?php
+namespace Application\BlogBundle\Admin;
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/CategoryAdmin.php]: Please, add newline before namespace here and in other files

File: src/Application/BlogBundle/Admin/PostAdmin.php [hunk 20]
```diff
@@ -1,9 +1,10 @@
-<?php namespace Application\BlogBundle\Admin;
+<?php
+namespace Application\BlogBundle\Admin;
 
 use Sonata\AdminBundle\Admin\Admin;
-use Sonata\AdminBundle\Datagrid\DatagridMapper;
 use Sonata\AdminBundle\Datagrid\ListMapper;
 use Sonata\AdminBundle\Form\FormMapper;
+use Application\BlogBundle\Form\Type\ThumbnailType;
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/PostAdmin.php]: Please, group uses

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        $this->createFolderIfNotExists();
+        $file = $uploadedFile;
+        try {
+            $file->move($this->folder, $file->getClientOriginalName());
+        } catch (FileException $e) {
+            return false;
+        }
+
+        return (string)$this->uploadDirectory . '/' . $file->getClientOriginalName();
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Use `DIRECTORY_SEPARATOR` here

File: src/Application/BlogBundle/Service/Thumbnail.php [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     * @param UploadedFile $uploadedFile
+     *
+     * @return bool|\Exception|FileException
+     */
+    public function save(UploadedFile $uploadedFile)
+    {
+        $this->createFolderIfNotExists();
+        $file = $uploadedFile;
```
Comment (aivus) [inline src/Application/BlogBundle/Service/Thumbnail.php]: Why you don't use `$uploadedFile` variable?

Comment (Madlexx) [inline src/Application/BlogBundle/Service/Thumbnail.php]: I forgot edit this(

**Pull request merged**